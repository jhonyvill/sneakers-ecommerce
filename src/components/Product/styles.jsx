import styled from "styled-components";

export const Container = styled.section`
  display: grid;
  grid-template-columns: 1fr 1fr;
  align-items: center;
  padding: 4rem 2rem;
  gap: 1rem 4rem;
  box-sizing: border-box;

  @media (max-width: ${({ theme }) => theme.screen.tablet}) {
    grid-template-columns: 1fr;
    padding: 0;
    gap: 2rem 0;
  }

  @media (max-width: ${({ theme }) => theme.screen.largeMobile}) {
    padding: 0;
  }
`;

export const ImagesContainer = styled.div`
  max-width: 29rem;
  justify-self: center;

  @media (max-width: ${({ theme }) => theme.screen.tablet}) {
    max-width: 100%;
  }
`;

export const DetailsContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1rem 0;
  max-width: 28.125rem;
  justify-self: center;

  @media (max-width: ${({ theme }) => theme.screen.tablet}) {
    max-width: 600px;
    margin: 0 auto;
  }
  @media (max-width: ${({ theme }) => theme.screen.smallTablet}) {
    max-width: 500px;
  }

`;
export const ProductActionContainer = styled.div`
  grid-column: 2;
  display: flex;
  gap: 1rem 0.8rem;

  @media (max-width: ${({ theme }) => theme.screen.tablet}) {
    grid-column: 1;
  }
  @media (max-width: ${({ theme }) => theme.screen.largeMobile}) {
    flex-direction: column;
  }
`;