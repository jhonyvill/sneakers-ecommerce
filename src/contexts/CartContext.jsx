import React from "react";
import { useReducer } from "react";

export const CartContext = React.createContext();

export const CartProvider = ({ children }) => {
  const initialState = { items: [] };

  function cartReducer(state, action) {
    switch (action.type) {
      case "ADD_ITEM":
        return { ...state, items: [...state.items, action.payload] };
      case "UPDATE_ITEM":
        const { product } = action.payload;
        const itemIndex = state.items.findIndex(
          (item) => item.product.id === product.id
        );

        const updatedItems = [...state.items];
        updatedItems.slice(itemIndex, product);
        return { ...state, items: updatedItems };
      default:
        state;
    }
  }

  const [state, dispatch] = useReducer(cartReducer, initialState);

  return (
    <>
      <CartContext.Provider
        value={{ cartState: state, cartDispatch: dispatch }}
      >
        {children}
      </CartContext.Provider>
    </>
  );
};
