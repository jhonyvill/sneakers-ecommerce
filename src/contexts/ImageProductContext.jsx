import React from "react";
import { useState } from "react";

export const ImageProductContext = React.createContext();

export const ImageProductProvider = ({ children }) => {
  const [selectedImage, setSelectedImage] = useState({
    image: null,
    index: 0,
  });

  const handleSelectedModalImage = (image, index) => {
    setSelectedImage({ image, index });
  };

  return (
    <>
      <ImageProductContext.Provider
        value={{
          imageSelected: selectedImage.image,
          indexImageSelected: selectedImage.index,
          handleSelectedModalImage,
        }}
      >
        {children}
      </ImageProductContext.Provider>
    </>
  );
};
