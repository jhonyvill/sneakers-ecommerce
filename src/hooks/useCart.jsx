import React from "react";
import { useState, useContext } from "react";
import { CartContext } from "../contexts/CartContext";

export const useCart = () => {
  const { cartState, cartDispatch } = useContext(CartContext);
  const [quantitySelected, setQuantitySelected] = useState(0);
  const cartHasItems = cartState.items.length > 0;

  function findProductInCart(product) {
    return cartState.items.find((item) => item.product.id === product.id);
  }

  function addCartProduct(product) {
    const existentProduct = findProductInCart(product);
    if(existentProduct){
      cartDispatch({
        type: "UPDATE_ITEM",
        payload: {
          product: product,
          quantity: (existentProduct.quantity += quantitySelected),
        },
      });
    }else if(quantitySelected > 0){
      cartDispatch({
        type: "ADD_ITEM",
        payload: { product: product, quantity: quantitySelected },
      });
    }
    setQuantitySelected(0);
  }

  return {
    quantitySelected,
    setQuantitySelected,
    addCartProduct,
    cartHasItems
  };
};
