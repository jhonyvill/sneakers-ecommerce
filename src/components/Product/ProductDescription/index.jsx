import * as S from "./styles";
import PropTypes from "prop-types";
import { useProductPrice } from "../../../hooks/useProductPrice";

const ProductDescription = ({ product }) => {
  const { price, finalPrice, discountPercent, hasDiscount } =
    useProductPrice(product);

  return (
    <S.Container aria-label="Descrição do produto">
      <S.Brand>{product.brand}</S.Brand>
      <S.Title>{product.title}</S.Title>
      <S.Description>{product.description}</S.Description>

      {hasDiscount ? (
        <S.PriceContainer>
          <S.FinalPriceContainer>
            <S.FinalPrice>${finalPrice}</S.FinalPrice>
            <S.Discount>{discountPercent}%</S.Discount>
          </S.FinalPriceContainer>

          <S.Price>${price}</S.Price>
        </S.PriceContainer>
      ) : (
        <S.FinalPrice>${price}</S.FinalPrice>
      )}
    </S.Container>
  );
};

ProductDescription.propTypes = {
  product: PropTypes.object.isRequired,
};

export default ProductDescription;
