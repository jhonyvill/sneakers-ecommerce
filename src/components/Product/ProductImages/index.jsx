import React from "react";
import * as S from "./styles";
import NextSvg from "../../../assets/icon-next.svg?react";
import PreviousSvg from "../../../assets/icon-previous.svg?react";
import PropTypes from "prop-types";
import { useMedia } from "../../../hooks/useMedia";
import { useProductImage } from "../../../hooks/useProductImage";

const ProductImages = ({ images, modalActive, handleModalActive }) => {
  const {
    featuredImage,
    imageIndexActive,
    handleChangeImage,
    handleClickFeaturedImage,
  } = useProductImage(images);

  const smallScreen = useMedia("(max-width: 820px)");
  
  return (
    <S.Container>
      <S.FeaturedImageContainer
        $container={(smallScreen && modalActive) ? "small" : "normal"}
      >
        <S.FeaturedImage
          width="1000px"
          height="1000px"
          src={featuredImage.src}
          alt={featuredImage.alt}
          onClick={() => {
            if (!modalActive) {
              handleClickFeaturedImage();
              handleModalActive(true);
            }
          }}
        />

        {(modalActive || smallScreen) && (
          <S.ButtonWrapper>
            <S.PreviousButton
              onClick={() => {
                const previousIndex = imageIndexActive - 1;
                handleChangeImage(images[previousIndex], previousIndex);
              }}
            >
              <PreviousSvg />
            </S.PreviousButton>
            <S.NextButton
              onClick={() => {
                const nextIndex = imageIndexActive + 1;
                handleChangeImage(images[nextIndex], nextIndex);
              }}
            >
              <NextSvg />
            </S.NextButton>
          </S.ButtonWrapper>
        )}
      </S.FeaturedImageContainer>

      <S.ImagesGrid>
        {images.map((image, index) => {
          return (
            <S.Image
              key={index}
              className={imageIndexActive === index && "active"}
              onClick={() => handleChangeImage(image, index)}
            >
              <img
                width="176px"
                height="176px"
                src={image.thumb}
                alt={image.alt}
              />
            </S.Image>
          );
        })}
      </S.ImagesGrid>
    </S.Container>
  );
};

ProductImages.propTypes = {
  images: PropTypes.array.isRequired,
  modalActive: PropTypes.bool,
  handleModalActive: PropTypes.func,
};

export default ProductImages;
