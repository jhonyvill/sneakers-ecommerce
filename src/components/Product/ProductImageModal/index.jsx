import React from "react";
import * as S from "./styles";
import CloseSvg from "../../../assets/icon-close.svg?react";
import PropTypes from "prop-types";
import ProductImages from "../ProductImages";

const ProductImageModal = ({ images, handleModalActive }) => {
  return (
    <S.Container aria-label="modal imagens do produto">
      <S.Modal>
        <S.CloseButton data-testid='close-button' onClick={() => handleModalActive(false)}>
          <CloseSvg />
        </S.CloseButton>

        <S.ImageContainer>
          <ProductImages
            images={images}
            modalActive={true}
            handleModalActive={handleModalActive}
          />
        </S.ImageContainer>
      </S.Modal>
    </S.Container>
  );
};

ProductImageModal.protoTypes = {
  images: PropTypes.array.isRequired,
  handleModalActive: PropTypes.func.isRequired,
};

export default ProductImageModal;
