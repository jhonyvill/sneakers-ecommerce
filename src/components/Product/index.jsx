import React from "react";
import * as S from "./styles";
import CartSvg from "../../assets/icon-cart.svg?react";
import { useState, useEffect } from "react";
import { useFetch } from "../../hooks/useFetch";
import Button from "../Utils/Button";
import Counter from "../Utils/Counter";
import ProductImages from "./ProductImages";
import ProductImageModal from "./ProductImageModal";
import ProductDescription from "./ProductDescription";
import { useCart } from "../../hooks/useCart";

const Product = () => {
  const { data, request } = useFetch();
  const [modalActive, setModalActive] = useState(false);
  const { quantitySelected, setQuantitySelected, addCartProduct } = useCart();

  useEffect(() => {
    async function getPhotos() {
      request("./data.json");
    }
    getPhotos();
  }, [request]);

  if (!data) return null;
  return (
    <S.Container>
      <S.ImagesContainer>
        <ProductImages
          images={data.pictures}
          modalActive={false}
          handleModalActive={setModalActive}
        />
      </S.ImagesContainer>

      <S.DetailsContainer>
        <ProductDescription product={data} />

        <S.ProductActionContainer>
          <Counter count={quantitySelected} setCount={setQuantitySelected} />
          <Button
            className={"shadowActive"}
            aria-label="adicionar ao carrinho"
            onClick={() => addCartProduct(data)}
          >
            <CartSvg /> Add to cart
          </Button>
        </S.ProductActionContainer>
      </S.DetailsContainer>

      {modalActive && (
        <ProductImageModal
          images={data.pictures}
          handleModalActive={setModalActive}
        />
      )}
    </S.Container>
  );
};

export default Product;
