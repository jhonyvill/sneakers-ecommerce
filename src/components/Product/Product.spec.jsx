import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderWithTheme } from "../../../.jest/utils";
import { CartProvider } from "../../contexts/CartContext";
import { ImageProductProvider } from "../../contexts/ImageProductContext";
import * as useFetchHook from "../../hooks/useFetch";
import * as hookCart from "../../hooks/useCart";
import Product from ".";

function renderProduct() {
  return renderWithTheme(
    <CartProvider>
      <ImageProductProvider>
        <Product />
      </ImageProductProvider>
    </CartProvider>
  );
}

function createUseFetchSpy(){
  return jest.spyOn(useFetchHook, 'useFetch').mockImplementationOnce(() => ({
    data: null,
    request: jest.fn()
  }));
}
function createUseCartSpy(quantityItems, mockFuntion = jest.fn()){
  return jest.spyOn(hookCart, 'useCart').mockReturnValue({
    quantitySelected: quantityItems,
    addCartProduct: mockFuntion
  });
}

jest.mock("../../hooks/useMedia", () => ({
  __esModule: true,
  useMedia: () => {
    return false;
  },
}));

jest.mock("../../hooks/useFetch", () => ({
  __esModule: true,
  useFetch: () => ({
    data: {
      id: "id-1",
      brand: "brand-1",
      title: "title-1",
      price: 10,
      discount: 0.5,
      description: "description-1",
      pictures: [
        {
          src: "image-1.jpg",
          thumb: "thumbnail 1.jpg",
          alt: "product 1",
        },
        {
          src: "image-2.jpg",
          thumb: "thumbnail 2.jpg",
          alt: "product 2",
        },
        {
          src: "image-3.jpg",
          thumb: "thumbnail 3.jpg",
          alt: "product 3",
        },
        {
          src: "image-4.jpg",
          thumb: "thumbnail 4.jpg",
          alt: "product 4",
        },
      ],
    },
    request: jest.fn(),
  }),
}));

describe("<Product />", () => {

  it("should render Product component", async () => {
    renderProduct();

    expect.assertions(5);
    expect(screen.getByRole('button', {name: /add to cart/i})).toBeInTheDocument();
    expect(screen.getByRole('region', {name: 'imagens do produto'})).toBeInTheDocument();
    expect(screen.getByRole('region', {name: /descrição do produto/i})).toBeInTheDocument();
    expect(screen.getByRole('region', {name: 'selecione a quantidade de itens'})).toBeInTheDocument();
    expect(screen.queryByRole('region', {name: 'modal imagens do produto'})).not.toBeInTheDocument();
  });
  it("should render Product component with modal active", async () => {
    const { getByTestId } = renderProduct();

    const featuredImage = getByTestId("featured-image");
    await userEvent.click(featuredImage);

    expect.assertions(1);
    expect(screen.queryByRole('region', {name: 'modal imagens do produto'})).toBeInTheDocument();
  });

  it("should render empty div when data is null", () => {
    createUseFetchSpy();

    renderProduct();
    expect.assertions(5);
    expect(screen.queryByRole('button', {name: /add to cart/i})).not.toBeInTheDocument();
    expect(screen.queryByRole('region', {name: 'imagens do produto'})).not.toBeInTheDocument();
    expect(screen.queryByRole('region', {name: /descrição do produto/i})).not.toBeInTheDocument();
    expect(screen.queryByRole('region', {name: 'selecione a quantidade de itens'})).not.toBeInTheDocument();
    expect(screen.queryByRole('region', {name: 'modal imagens do produto'})).not.toBeInTheDocument();
  });

  it("should call 'addCartProduct' function when addToCart button clicked", async () => {
    const mockFuntion = jest.fn();
    createUseCartSpy(0, mockFuntion);

    renderProduct();

    const button = screen.getByRole('button', {name: /add to cart/i});
    await userEvent.click(button);

    expect.assertions(1);
    expect(mockFuntion).toHaveBeenCalledTimes(1);
  });
});

describe("<Product /> snapshots", () => {
  it("should match snapshot", () => {
    const { container } = renderProduct();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should match snapshot when modal active", async() => {
    const { container } = renderProduct();
    const featuredImage = screen.getByTestId("featured-image");
    await userEvent.click(featuredImage);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should match snapshot when data is null", async() => {
    createUseFetchSpy();
    const { container } = renderProduct();
    expect(container.firstChild).toMatchSnapshot();
  });
});
