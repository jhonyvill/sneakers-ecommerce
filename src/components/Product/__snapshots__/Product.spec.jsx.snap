// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> snapshots should match snapshot 1`] = `
<section
  class="styles__Container-sc-11z2e6s-0 cvKyIH"
>
  <div
    class="styles__ImagesContainer-sc-11z2e6s-1 kTpCiU"
  >
    <section
      aria-label="imagens do produto"
      class="styles__Container-sc-1rfz8de-0 cjexoL"
    >
      <div
        class="styles__FeaturedImageContainer-sc-1rfz8de-1 hkalnS"
        data-testid="container-featured-image"
      >
        <img
          alt="product 1"
          class="styles__FeaturedImage-sc-1rfz8de-2 hLfEin"
          data-testid="featured-image"
          height="1000px"
          src="image-1.jpg"
          width="1000px"
        />
      </div>
      <div
        class="styles__ImagesGrid-sc-1rfz8de-3 gtpYQD"
      >
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl active"
          data-testid="grid-image-0"
        >
          <img
            alt="product 1"
            height="176px"
            src="thumbnail 1.jpg"
            width="176px"
          />
        </div>
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl"
          data-testid="grid-image-1"
        >
          <img
            alt="product 2"
            height="176px"
            src="thumbnail 2.jpg"
            width="176px"
          />
        </div>
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl"
          data-testid="grid-image-2"
        >
          <img
            alt="product 3"
            height="176px"
            src="thumbnail 3.jpg"
            width="176px"
          />
        </div>
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl"
          data-testid="grid-image-3"
        >
          <img
            alt="product 4"
            height="176px"
            src="thumbnail 4.jpg"
            width="176px"
          />
        </div>
      </div>
    </section>
  </div>
  <div
    class="styles__DetailsContainer-sc-11z2e6s-2 haQqEv"
  >
    <section
      aria-label="descrição do produto"
      class="styles__Container-sc-ypft3n-0 hhjPNM"
    >
      <p
        class="styles__Brand-sc-ypft3n-1 eBdmJx"
      >
        brand-1
      </p>
      <h1
        class="styles__Title-sc-ypft3n-2 kFUCun"
      >
        title-1
      </h1>
      <p
        class="styles__Description-sc-ypft3n-3 jIbbAQ"
      >
        description-1
      </p>
      <div
        class="styles__PriceContainer-sc-ypft3n-4 jUenDh"
      >
        <span
          class="styles__FinalPriceContainer-sc-ypft3n-5 hXifzc"
        >
          <span
            class="styles__FinalPrice-sc-ypft3n-6 cInfIl"
          >
            $
            5.00
          </span>
          <span
            class="styles__Discount-sc-ypft3n-7 cyFKTa"
          >
            50
            %
          </span>
        </span>
        <p
          class="styles__Price-sc-ypft3n-8 gNjCoe"
        >
          $
          10.00
        </p>
      </div>
    </section>
    <div
      class="styles__ProductActionContainer-sc-11z2e6s-3 gMjTvO"
    >
      <section
        aria-label="selecione a quantidade de itens"
        class="styles__Container-sc-6vyuvp-0 klSogO"
      >
        <button
          aria-label="menos"
          class="styles__CounterButton-sc-6vyuvp-1 hNfiSo"
        >
          <icon-mock />
        </button>
        <span
          data-testid="counter"
        >
          0
        </span>
        <button
          aria-label="mais"
          class="styles__CounterButton-sc-6vyuvp-1 hNfiSo"
        >
          <icon-mock />
        </button>
      </section>
      <button
        aria-labelledby="add-btn"
        class="styles__StyledButton-sc-1csa66j-0 hmYbLP shadowActive"
      >
        <icon-mock />
         
        <span
          id="add-btn"
        >
          Add to cart
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`<Product /> snapshots should match snapshot when data is null 1`] = `null`;

exports[`<Product /> snapshots should match snapshot when modal active 1`] = `
<section
  class="styles__Container-sc-11z2e6s-0 cvKyIH"
>
  <div
    class="styles__ImagesContainer-sc-11z2e6s-1 kTpCiU"
  >
    <section
      aria-label="imagens do produto"
      class="styles__Container-sc-1rfz8de-0 cjexoL"
    >
      <div
        class="styles__FeaturedImageContainer-sc-1rfz8de-1 hkalnS"
        data-testid="container-featured-image"
      >
        <img
          alt="product 1"
          class="styles__FeaturedImage-sc-1rfz8de-2 hLfEin"
          data-testid="featured-image"
          height="1000px"
          src="image-1.jpg"
          width="1000px"
        />
      </div>
      <div
        class="styles__ImagesGrid-sc-1rfz8de-3 gtpYQD"
      >
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl active"
          data-testid="grid-image-0"
        >
          <img
            alt="product 1"
            height="176px"
            src="thumbnail 1.jpg"
            width="176px"
          />
        </div>
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl"
          data-testid="grid-image-1"
        >
          <img
            alt="product 2"
            height="176px"
            src="thumbnail 2.jpg"
            width="176px"
          />
        </div>
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl"
          data-testid="grid-image-2"
        >
          <img
            alt="product 3"
            height="176px"
            src="thumbnail 3.jpg"
            width="176px"
          />
        </div>
        <div
          class="styles__Image-sc-1rfz8de-4 gGfERl"
          data-testid="grid-image-3"
        >
          <img
            alt="product 4"
            height="176px"
            src="thumbnail 4.jpg"
            width="176px"
          />
        </div>
      </div>
    </section>
  </div>
  <div
    class="styles__DetailsContainer-sc-11z2e6s-2 haQqEv"
  >
    <section
      aria-label="descrição do produto"
      class="styles__Container-sc-ypft3n-0 hhjPNM"
    >
      <p
        class="styles__Brand-sc-ypft3n-1 eBdmJx"
      >
        brand-1
      </p>
      <h1
        class="styles__Title-sc-ypft3n-2 kFUCun"
      >
        title-1
      </h1>
      <p
        class="styles__Description-sc-ypft3n-3 jIbbAQ"
      >
        description-1
      </p>
      <div
        class="styles__PriceContainer-sc-ypft3n-4 jUenDh"
      >
        <span
          class="styles__FinalPriceContainer-sc-ypft3n-5 hXifzc"
        >
          <span
            class="styles__FinalPrice-sc-ypft3n-6 cInfIl"
          >
            $
            5.00
          </span>
          <span
            class="styles__Discount-sc-ypft3n-7 cyFKTa"
          >
            50
            %
          </span>
        </span>
        <p
          class="styles__Price-sc-ypft3n-8 gNjCoe"
        >
          $
          10.00
        </p>
      </div>
    </section>
    <div
      class="styles__ProductActionContainer-sc-11z2e6s-3 gMjTvO"
    >
      <section
        aria-label="selecione a quantidade de itens"
        class="styles__Container-sc-6vyuvp-0 klSogO"
      >
        <button
          aria-label="menos"
          class="styles__CounterButton-sc-6vyuvp-1 hNfiSo"
        >
          <icon-mock />
        </button>
        <span
          data-testid="counter"
        >
          0
        </span>
        <button
          aria-label="mais"
          class="styles__CounterButton-sc-6vyuvp-1 hNfiSo"
        >
          <icon-mock />
        </button>
      </section>
      <button
        aria-labelledby="add-btn"
        class="styles__StyledButton-sc-1csa66j-0 hmYbLP shadowActive"
      >
        <icon-mock />
         
        <span
          id="add-btn"
        >
          Add to cart
        </span>
      </button>
    </div>
  </div>
  <section
    aria-label="modal imagens do produto"
    class="styles__Container-sc-tc5yrj-0 iYchpi"
  >
    <div
      class="styles__Modal-sc-tc5yrj-1 fuToiU"
    >
      <button
        class="styles__CloseButton-sc-tc5yrj-3 fscaYl"
        data-testid="close-button"
      >
        <icon-mock />
      </button>
      <div
        class="styles__ImageContainer-sc-tc5yrj-2 hlTyuA"
      >
        <section
          aria-label="imagens do produto"
          class="styles__Container-sc-1rfz8de-0 cjexoL"
        >
          <div
            class="styles__FeaturedImageContainer-sc-1rfz8de-1 hkalnS"
            data-testid="container-featured-image"
          >
            <img
              alt="product 1"
              class="styles__FeaturedImage-sc-1rfz8de-2 hLfEin"
              data-testid="featured-image"
              height="1000px"
              src="image-1.jpg"
              width="1000px"
            />
            <div
              class="styles__ButtonWrapper-sc-1rfz8de-5 hFyXwb"
            >
              <button
                aria-label="imagem anterior"
                class="styles__ModalButton-sc-1rfz8de-6 styles__PreviousButton-sc-1rfz8de-7 bobShG kxWovx"
              >
                <icon-mock />
              </button>
              <button
                aria-label="próxima imagem"
                class="styles__ModalButton-sc-1rfz8de-6 styles__NextButton-sc-1rfz8de-8 bobShG gUtcMk"
              >
                <icon-mock />
              </button>
            </div>
          </div>
          <div
            class="styles__ImagesGrid-sc-1rfz8de-3 gtpYQD"
          >
            <div
              class="styles__Image-sc-1rfz8de-4 gGfERl active"
              data-testid="grid-image-0"
            >
              <img
                alt="product 1"
                height="176px"
                src="thumbnail 1.jpg"
                width="176px"
              />
            </div>
            <div
              class="styles__Image-sc-1rfz8de-4 gGfERl"
              data-testid="grid-image-1"
            >
              <img
                alt="product 2"
                height="176px"
                src="thumbnail 2.jpg"
                width="176px"
              />
            </div>
            <div
              class="styles__Image-sc-1rfz8de-4 gGfERl"
              data-testid="grid-image-2"
            >
              <img
                alt="product 3"
                height="176px"
                src="thumbnail 3.jpg"
                width="176px"
              />
            </div>
            <div
              class="styles__Image-sc-1rfz8de-4 gGfERl"
              data-testid="grid-image-3"
            >
              <img
                alt="product 4"
                height="176px"
                src="thumbnail 4.jpg"
                width="176px"
              />
            </div>
          </div>
        </section>
      </div>
    </div>
  </section>
</section>
`;
